typescript/sdk/src/metadata/blockExplorer.ts

```diff
@@ -23,8 +52,20 @@ export function getExplorerApi(
   // TODO solana + cosmos support here as needed
   if (protocol !== ProtocolType.Ethereum) return null;
   if (!blockExplorers?.length || !blockExplorers[index].apiUrl) return null;
+
+  // Convert to V2 format if it's an Etherscan-like API
+  const chainId =
+    typeof metadata.chainId === 'string'
+      ? parseInt(metadata.chainId)
+      : metadata.chainId;
+  const convertedApiUrl = convertToEtherscanV2Url(
+    blockExplorers[index].apiUrl,
+    chainId,
+    blockExplorers[index].family,
+  );
+
   return {
-    apiUrl: blockExplorers[index].apiUrl,
+    apiUrl: convertedApiUrl,
```

Comment:
what about etherscan-like apis that haven't upgraded to v2? (like some long tails)